### PR TITLE
Fix various highlighting errors

### PR DESCRIPTION
**Pull request by DaelonSuzuka**
This PR will have a collection of syntax highlighting improvements, starting with issues outlined in #379, 

- [x] `in` keyword

image

- [x] `await` and `super`

Code_rfH2Tv7ExD

- [x] `get` and `set` (harder)

Code_fffonCRmJX

- [x] `@annotation` syntax

I'm not sure if the list of "decorator" keywords is exhaustive, but the basic mechanism definitely works.

image

- [x] investigate other new Godot 4 features
- [x] lambdas

Code_uHjnckmxFp

- [x] type arrays

image

- [x] `trait` and `namespace`

### CODE REVIEW

**Comment by DaelonSuzuka** [comment]
@Calinou I couldn't find any other new GDScript features that were missing syntax highlighting.

Let me know if you hate any of my colorization choices. I was torn on giving `@annotations` a more exotic color, but I actually think it looks fine like this.

**Comment by Calinou** [comment]
> Let me know if you hate any of my colorization choices. I was torn on giving @annotations a more exotic color, but I actually think it looks fine like this.

Does VS Code have a standard annotation color (e.g. the one used in Python, or for preprocessor keywords in C++)?

File: syntaxes/GDScript.tmLanguage.json [hunk 1]
```diff
@@ -807,4 +894,4 @@
             ]
         }
     }
-}
+}
```
Comment (Calinou) [inline syntaxes/GDScript.tmLanguage.json:897]:
```suggestion
}

```

File: syntaxes/examples/gdscript2.gd [hunk 2]
```diff
@@ -0,0 +1,55 @@
+extends Node
```
Comment (Calinou) [inline syntaxes/examples/gdscript2.gd:1]: I think it'd be good idea to submit this code as a GDScript integration test in godotengine/godot :slightly_smiling_face: 

I can look into doing that.

**Review by Calinou — APPROVED** [review]
Thanks!

We can change the annotation color in a future PR.

File: syntaxes/examples/gdscript2.gd (hunk 2 above)
Comment (DaelonSuzuka) [reply syntaxes/examples/gdscript2.gd:1]: Is there an ergonomic way to automate a highlighting test? I haven't been able to think of one that wouldn't be a massive pain to define the success/failure conditions.

I think finding a way to share test files with the engine is a great idea, though. Having files that we know exercise all of the compiler's features would really increase confidence that we're covering all the possible syntax.

**Comment by DaelonSuzuka** [comment]
> Does VS Code have a standard annotation color (e.g. the one used in Python, or for preprocessor keywords in C/C++)?

The current annotation color is the same as python's decorators. Godot's script editor colors the annotations like keywords, but I thought that didn't look quite as good here. C/C++ preprocessor things are mostly just blue like keywords, as far as I remember. I know rust has some complicated syntax, I can see what they do for meta-informational markings like this. C# and C++ might also have something similar.

**Pull request merged**